crates/ruff_python_formatter/src/expression/expr_await.rs

```diff
@@ -15,24 +15,27 @@ impl FormatNodeRule<ExprAwait> for FormatExprAwait {
     fn fmt_fields(&self, item: &ExprAwait, f: &mut PyFormatter) -> FormatResult<()> {
         let ExprAwait { range: _, value } = item;
 
-        let format_value = format_with(|f: &mut PyFormatter| {
-            if f.context().node_level().is_parenthesized() {
-                value.format().fmt(f)
-            } else {
-                maybe_parenthesize_expression(value, item, Parenthesize::Optional).fmt(f)
-            }
-        });
-
-        write!(f, [text("await"), space(), format_value])
+        write!(
+            f,
+            [
+                text("await"),
+                space(),
+                maybe_parenthesize_expression(value, item, Parenthesize::IfRequired)
```

Comment:
I introduced this for the with handling and it gives us the nice property that it goes through `NeedsParentheses`. Meaning, it respects whether the expressions wants to be parenthesized or not.